h5py/h5d.pyx

```diff
@@ -398,12 +399,19 @@ cdef class DatasetID(ObjectID):
 
     IF HDF5_VERSION >= (1, 8, 11):
 
-        def write_direct_chunk(self, offsets, bytes data, H5Z_filter_t filter_mask=H5Z_FILTER_NONE, PropID dxpl=None):
-            """ (offsets, bytes data, H5Z_filter_t filter_mask=H5Z_FILTER_NONE, PropID dxpl=None)
+        def write_direct_chunk(self, offsets, bytes data, filter_mask=0x00000000, PropID dxpl=None):
+            """ (offsets, bytes data, uint32_t filter_mask=0x00000000, PropID dxpl=None)
 
             Writes data from a bytes array (as provided e.g. by struct.pack) directly
             to a chunk at position specified by the offsets argument.
 
+            The `filter_mask` is a bit field of up to 32 values. It specify
+            which filters in the pipeline have to be disabled to decompress the
+            `data`.
+            Using the default `0x00000000` with use all the filters when the
+            data will be read, while `0xFFFFFFFF` will skip all the filters
```

Comment:
I'd still like this to be a bit clearer. Something like:

> The `write_direct_chunk` function bypasses any filters HDF5 would normally apply to written data. However, calling code may apply filters (e.g. gzip compression) itself before writing the data. `filter_mask` records which filters have been applied to this chunk, of the filter pipeline defined for that dataset. Each bit set to 1 means that the filter in the corresponding position in the pipeline was *not* applied. So the default value of 0 means that all defined filters have been applied to the data before calling this function.

And something similar for `read_direct_chunk`.

Reply:
As you like